pkg/daemon/ceph/client/upgrade.go

```diff
@@ -217,8 +227,9 @@ func okToStopDaemon(context *clusterd.Context, clusterInfo *ClusterInfo, deploym
 // okToContinueOSDDaemon determines whether it's fine to go to the next osd during an upgrade
 // This basically makes sure all the PGs have settled
 func okToContinueOSDDaemon(context *clusterd.Context, clusterInfo *ClusterInfo) error {
-	// Reconciliating PGs should not take too long so let's wait up to 10 minutes
-	err := util.Retry(10, 60*time.Second, func() error {
+	// for OSD, the delay between each retry should be waitTimeoutForHealthyOSD/maxRetries seconds
+	delay := clusterInfo.OsdUpgradeTimeout / maxRetries
```

Comment:
The timing is currently up to 1 minute for ok-to-stop, and 5 minutes (default) for ok-to-continue, right? This seems backwards to me. What about waiting 5 min for ok-to-stop and 1 min for ok-to-continue? If there are more OSDs to upgrade after this one, they will wait 5 min to upgrade with the ok-to-stop check.

Reply:
@leseb Remind me why we even check for ok-to-continue? Seems like we only need to wait if we're checking for ok-to-stop and update an osd. If we continue as soon as the new pod is running, the next OSD will wait anyway on the ok-to-stop, right?

Reply:
With latest changes in the PR, we having following timeout and retry configurations:
1. ok-to-stop
   - OSD:
         - retries = waitTimeoutForHealthyOSD(in seconds)/10
         - delay = 10 seconds. 
         - For example, if user gives waitTimeoutForHealthyOSD as 10 (600 seconds), then, retry count is 60, delay between 
           each retry is 10 seconds.
           
   - Other daemons:
        - retries = 10
        - delay =  1 minute
        - Total = 10 minutes

2  ok-to-continue
   - OSD:  same as ok-to-stop above
   - MDS: 
      - retries = 10
      - delay = 15 seconds
      - Total = 150 seconds. 
   - ok-to-continue is not configured for any other daemons other than OSD and MDS

Reply:
Removed `ok-to-continue` logic for OSDs